src/hidpp20.c

```diff
@@ -1129,6 +1129,11 @@ hidpp20_onboard_profiles_read_sector(struct hidpp20_device *device,
 	count = sector_size;
 
 	for (offset = 0; offset < sector_size; offset += 16) {
+		/*
+		 * the firmware replies with an ERR_INVALID_ARGUMENT error
+		 * if we try to read past sector_size - 16, so when we are left with
+		 * less than 16 bytes to read we need to read from sector_size - 16
+		 */
```

Comment:
for the commit message: please use present tense, because you're describing what the patch does when applied. and a hidpp20 prefix please, so "hidpp20: add comment for clarification"

Also, can we just squash this one into the previous commit please?